rampwf/prediction_types/regression.py

```diff
@@ -10,17 +10,28 @@
 from .base import BasePrediction
 
 
-class Predictions(BasePrediction):
-
-    def __init__(self, labels=None, y_pred=None, y_true=None, shape=None):
-        self.labels = labels
-        if y_pred is not None:
-            self.y_pred = np.array(y_pred)
-        elif y_true is not None:
-            self.y_pred = np.array(y_true)
-        elif shape is not None:
-            self.y_pred = np.empty(shape, dtype=float)
-            self.y_pred.fill(np.nan)
+def _regression_init(self, y_pred=None, y_true=None, n_samples=None):
+    if y_pred is not None:
+        self.y_pred = np.array(y_pred)
+    elif y_true is not None:
+        self.y_pred = np.array(y_true)
+    elif n_samples is not None:
+        n_columns = len(self.label_names)
+        if n_columns == 0:
+            shape = (n_samples)
         else:
-            raise ValueError("Missing init argument: y_pred, y_true, f_name "
-                             "or n_samples")
+            shape = (n_samples, n_columns)
+        self.y_pred = np.empty(shape, dtype=float)
+        self.y_pred.fill(np.nan)
+    else:
+        raise ValueError(
+            'Missing init argument: y_pred, y_true, or n_samples')
+
+
+def make_predictions_type(label_names=[]):
```

Comment:
We can change this later. It's just to have a single syntax when we use it, e.g. 
https://github.com/ramp-kits/titanic/blob/master/problem.py

Predictions = rw.prediction_types.multiclass.make_predictions_type(
    label_names=_prediction_label_names)

It could be simplified, e.g. if we expose them in __init__ to

Predictions = rw.prediction_types.make_regression_type(
    label_names=_prediction_label_names)

Would you prefer this?